AWSIoTTests/AWSIoTDataTests.m

```diff
@@ -995,17 +995,25 @@ - (void)testUsernameMetaData {
     
     // Check state after adding additional fields
     NSDictionary<NSString *,NSString *> * metaData = @{@"foo": @"bar", @"clazz": @"2"};
-    [iotDataManager addUserMetaData: metaData];
+    [iotDataManager updateUserMetaData: metaData];
     NSString *expectedUserMetaData = [NSString stringWithFormat:@"?SDK=iOS&Version=%@&foo=bar&clazz=2", AWSIoTSDKVersion];
     NSString *actualUserMetaData = [[iotDataManager mqttClient] userMetaData];
     XCTAssertTrue([actualUserMetaData isEqualToString:expectedUserMetaData]);
     
     // Check state after adding additional fields twice
-    NSDictionary<NSString *,NSString *> * metaData2 = @{@"foo2": @"bar2"};
-    [iotDataManager addUserMetaData: metaData2];
-    NSString *expectedUserMetaData2 = [NSString stringWithFormat:@"?SDK=iOS&Version=%@&foo=bar&clazz=2&foo2=bar2", AWSIoTSDKVersion];
+    NSDictionary<NSString *,NSString *> * metaData2 = @{@"foo": @"bar2", @"foo3": @"bar3", @"foo4": @""};
+    [iotDataManager updateUserMetaData: metaData2];
+    NSString *expectedUserMetaData2 = [NSString stringWithFormat:@"?SDK=iOS&Version=%@&foo=bar2&foo4&clazz=2&foo3=bar3", AWSIoTSDKVersion];
     NSString *actualUserMetaData2 = [[iotDataManager mqttClient] userMetaData];
     XCTAssertTrue([actualUserMetaData2 isEqualToString:expectedUserMetaData2]);
+
+    // Check truncation logic
+    NSDictionary<NSString *,NSString *> * metaData3 = @{@"foo": @"bar2", @"foo3": @"bar3", @"foo4": @"", @"unusuallyLongKeynameExpectedToCauseTruncation":@"unusuallyLongValueExpectedToCauseTruncation",
```

Comment:
It would be better to limit the number of assertions in a single test case. Testing each member of a value might make sense in a single test case; testing validation logic doesn't really belong in the existence test.

More importantly, dictionary iteration is not guaranteed to have a deterministic order, so you can't rely on the metadata string always being the same unless there is logic that guarantees the order string. Which, if there is something that guarantees the order, needs its own test.